### DISCUSSION TITLE
How to style active tabs on mobile?

### DISCUSSION THREAD

**Discussion by martinszeltins** [tailwindlabs/tailwindcss#5581] [Help] [answered]
Let's say we have some tabs like this and we apply some classes if it is the active tab. But... what about if at different breakpoints we want to apply classes for both conditions - if this is the active tab AND at this breakpoint add these classes but if this is the active tab AND at another endpoint add different classes.

````
<ul>
  <li
    class="cursor-pointer py-2 px-4 text-gray-500 border-b-8 text-green-500 border-green-500"
    :class="{ 'text-green-500 border-green-500': isActiveTab }">
    Tab
  </li>
</ul>
````

Active tab & at sm breakpoint add these classes:
`text-green-500 border-green-500`

Active tab & at md breakpoint add these classes:
`text-green-500 border-green-500 pl-5 pb-2`

**Comment by dance2die** [comment, marked as answer]
You can ~~chain~~ stack variants together like `sm:active:text-green-800`.
**Demo**: https://play.tailwindcss.com/Cb05nhxqB2?layout=horizontal&size=926x720

![tw-active-variant](https://user-images.githubusercontent.com/8465237/134545245-e77eb48e-9d6a-4c89-ab66-08b804d85d0b.gif)


If you aren't using libraries such as React/Vue/Svelte to componentize, the class names can be hard to maintain.  
In that case you can use [@apply ](https://tailwindcss.com/docs/functions-and-directives#apply) to create utilities to shorten and maintain separately.

---

**Edit**: Seems like the term's "stacking" not "chaining": https://tailwindcss.com/docs/dark-mode#stacking-with-other-variants

**Reply by martinszeltins** [reply]
> You can ~chain~ stack variants together like `sm:active:text-green-800`.
> **Demo**: https://play.tailwindcss.com/Cb05nhxqB2?layout=horizontal&size=926x720
> 
> ![tw-active-variant](https://user-images.githubusercontent.com/8465237/134545245-e77eb48e-9d6a-4c89-ab66-08b804d85d0b.gif)
> 
> If you aren't using libraries such as React/Vue/Svelte to componentize, the class names can be hard to maintain.
> In that case you can use [@apply ](https://tailwindcss.com/docs/functions-and-directives#apply) to create utilities to shorten and maintain separately.
> 
> **Edit**: Seems like the term's "stacking" not "chaining": https://tailwindcss.com/docs/dark-mode#stacking-with-other-variants

@dance2die I was thinking more of when the element has the "active" class AND on different breakpoints - not the active css state.

**Reply by dance2die** [reply]
> @dance2die I was thinking more of when the element has the "active" class AND on different breakpoints - not the active css state.

Based on your code, you can add variants.  

```
<ul>
  <li
    class="cursor-pointer py-2 px-4 text-gray-500 border-b-8 text-green-500 border-green-500"
    :class="{ 'text-green-500 border-green-500 md:pl-5 md:pb-2': isActiveTab }">
    Tab
  </li>
</ul>
```

`'text-green-500 border-green-500` will be applied for `xs/sm` breakpoints, while `md:pl-5 md:pb-2` for `md/lg/xl` when the tab is active.

If you are adding an `.active` class, then I'd create an external CSS selector such as`li.active` to `@apply` the classes `text-green-500 border-green-500 md:pl-5 md:pb-2`.

**Reply by martinszeltins** [reply]
> > @dance2die I was thinking more of when the element has the "active" class AND on different breakpoints - not the active css state.
> 
> Based on your code, you can add variants.
> 
> ```
> <ul>
>   <li
>     class="cursor-pointer py-2 px-4 text-gray-500 border-b-8 text-green-500 border-green-500"
>     :class="{ 'text-green-500 border-green-500 md:pl-5 md:pb-2': isActiveTab }">
>     Tab
>   </li>
> </ul>
> ```
> 
> `'text-green-500 border-green-500` will be applied for `xs/sm` breakpoints, while `md:pl-5 md:pb-2` for `md/lg/xl` when the tab is active.
> 
> If you are adding an `.active` class, then I'd create an external CSS selector such as`li.active` to `@apply` the classes `text-green-500 border-green-500 md:pl-5 md:pb-2`.

Oh, yes I see. That's what I was looking for. Somehow I didn't think about adding the breakpoints for the active class like that. Thanks!

**Reply by dance2die** [reply]
> 'text-green-500 border-green-500` will be applied for xs/sm breakpoints

Sorry but a correction. it will be applied for ALL breakpoints. not just `xs/sm`.

> md:pl-5 md:pb-2 for md/lg/xl when the tab is active.

This still holds true.
